### PR TITLE
Refactor CR to be component based

### PR DESCRIPTION
**Pull request by pfmackin**
Refactor the Verrazzano CR to have a core section and a set of components.  This PR only has the components that are currently configurable.  For this pass, the schema of the component is pretty much the same as before.

Also updated the CR spec in the install.md file.

### CODE REVIEW

File: install.md [hunk 1]
```diff
@@ -305,3 +225,123 @@ On an OKE install, this may indicate that there is a missing ingress rule or rul
...
+## Components
+| Field | Type | Description | Required
+| --- | --- | --- | --- |
+| certManager | [CertManagerComponent](#certmanager-component) | The cert-manager component config.  | No | 
+| dns | [DNSComponent](#dns-component) | The DNS component config.  | No | 
+| ingress | [IngressComponent](#ingress-component) | The ingress component config. | No | 
+| istio | [IstioComponent](#istio-component) | The Istio component config. | No | 
+
+## CertManager Component
+| Field | Type | Required | Description
```
Comment (bobdonat) [inline install.md:278]: headings don't match.  the last 2 columns should be Desciption | Required

File: install.md [hunk 2]
```diff
@@ -305,3 +225,123 @@ On an OKE install, this may indicate that there is a missing ingress rule or rul
...
+| --- | --- | --- | --- |
+| certificate | [Certificate](#certificate) | The certificate config. | No |
+
+## Certificate
+| Field | Type | Required | Description
```
Comment (bobdonat) [inline install.md:283]: headings don't match.  the last 2 columns should be Desciption | Required

File: install.md [hunk 3]
```diff
@@ -305,3 +225,123 @@ On an OKE install, this may indicate that there is a missing ingress rule or rul
...
+| --- | --- | --- | --- |
+| acme | [Acme](#acme) | The Acme config.  Either `acme` or `ca` must be specified. | No |
+| ca | [CertificateAuthority](#CertificateAuthority) | The certificate authority config.  Either `acme` or `ca` must be specified. | No |
+
+## Acme
+| Field | Type | Required | Description
```
Comment (bobdonat) [inline install.md:289]: headings don't match.  the last 2 columns should be Desciption | Required

File: install.md [hunk 4]
```diff
@@ -305,3 +225,123 @@ On an OKE install, this may indicate that there is a missing ingress rule or rul
...
+| --- | --- | --- | --- |
+| `provider` | string | Name of the Acme provider. |  Yes | 
+| `emailAddress` | string | Email address of the user. |  Yes | 
+
+## CertificateAuthority
+| Field | Type | Required | Description
```
Comment (bobdonat) [inline install.md:295]: headings don't match.  the last 2 columns should be Desciption | Required

File: operator/internal/installjob/install_config.go [hunk 5]
```diff
@@ -220,12 +230,21 @@ func newXipIoInstallConfig(vz *installv1alpha1.Verrazzano) *InstallConfiguration
 		DNS: DNS{
 			Type: DNSTypeXip,
 		},
-		Ingress: getIngress(vz.Spec.Ingress),
+		Ingress: Ingress{
```
Comment (bobdonat) [inline operator/internal/installjob/install_config.go:233]: this looks like a merge problem.  this was changed to use getIngress(...) and that is being undone

File: operator/internal/installjob/install_config.go [hunk 6]
```diff
@@ -241,15 +260,24 @@ func newExternalDNSInstallConfig(vz *installv1alpha1.Verrazzano) *InstallConfigu
 		DNS: DNS{
 			Type: DNSTypeExternal,
 			External: &ExternalDNS{
-				Suffix: vz.Spec.DNS.External.Suffix,
+				Suffix: vz.Spec.Components.DNS.External.Suffix,
```
Comment (bobdonat) [inline operator/internal/installjob/install_config.go:263]: same comment. this looks like a merge problem. this was changed to use getIngress(...) and that is being undone

File: operator/internal/installjob/install_config.go [hunk 7]
```diff
@@ -241,15 +260,24 @@ func newExternalDNSInstallConfig(vz *installv1alpha1.Verrazzano) *InstallConfigu
...
+				IstioInstallArgs: getIngressArgs(vz.Spec.Components.Istio.IstioInstallArgs),
 			},
 		},
-		Ingress: getIngress(vz.Spec.Ingress),
 		Certificates: Certificate{
 			IssuerType: CertIssuerTypeCA,
 			CA: &CertificateCA{
-				ClusterResourceNamespace: getCAClusterResourceNamespace(vz.Spec.Certificate.CA),
-				SecretName:               getCASecretName(vz.Spec.Certificate.CA),
+				ClusterResourceNamespace: "cattle-system",
```
Comment (bobdonat) [inline operator/internal/installjob/install_config.go:279]: this looks to be merge problem too.  This code was merged yesterday to not hard code the values

File: operator/internal/installjob/install_config.go [hunk 8]
```diff
@@ -323,18 +351,24 @@ func getIngressType(ingressType installv1alpha1.IngressType) IngressType {
 	return IngressTypeNodePort
 }
 
-// getIngress returns the json representation for the ingress
```
Comment (bobdonat) [inline operator/internal/installjob/install_config.go:326]: This funtion was incorrectly deleted.

File: operator/internal/installjob/install_config.go (hunk 7 above)
Comment (bobdonat) [reply operator/internal/installjob/install_config.go:279]: should be using getCAClusterResourceNamespace and getCASecretName

Comment (bobdonat) [reply operator/internal/installjob/install_config.go:252]: This is still a problem.  This is undoing a merge I recently did where hardcoded values were being used.

Comment (bobdonat) [reply operator/internal/installjob/install_config.go:252]: both fields

Comment (pfmackin) [reply operator/internal/installjob/install_config.go]: fixed

**Review by bobdonat — DISMISSED** [review]

**Review by mgianatagh — APPROVED** [review]
Re-approving for some small changes Paul had to push.

**Pull request merged**